### ISSUE TITLE
Schema.org

### ISSUE THREAD

**Issue by ways2read** [ways2read/InDesignA11Y#19]
Knowledge base reference: http://kb.daisy.org/publishing/docs/metadata/schema.org/index.html

**Comment by LauraB7** [comment]
User story: "As an end user, librarian or teacher I can find out about the accessibility features/hazards of a publication in the retail or library catalog." "As a publisher I can meet the EPUB accessibility specification by including accessibility metadata."

**Comment by raman211** [comment]
Hi @ways2read , @LauraB7 , @gregoriopellegrino 

This is Ramanjulu from Adobe InDesign Team. We are looking to give content creators a way to add the above accessibility and need some clarity from you all.

We have the proposed design specs available here https://xd.adobe.com/view/f296cb1f-2874-4e2f-83d4-2cf6fdc06794-a7c5/

Could you please share your feedback on the proposed designs for the same?
Thanks.

**Comment by LauraB7** [comment]
Writing on behalf of the group, we have a number of comments. 

1. Is it safe to presume that this metadata menu is available for reflowable EPUB as well as fixed-layout? The example shows the export options for FXL so we want to be sure. 
2. Some of these values are not obvious. Would it be possible to link to the [Daisy Knowledge Base ](http://kb.daisy.org/publishing/docs/metadata/)for explainers somewhere on this screen?
3. We note that there are tiny (i) buttons in two places and are curious what the pop-up window says. This might be the right place to put a link to the knowledge base. 
4. It would make sense to put the most common accessibilityFeature values at the top of the drop-down. 
5. Accessibility metadata is dynamic and constantly being worked on in standards groups. Will there be an easy way to update the options from time to time?
6. Given the last point, it might make sense to have a write-in field option.
7. Is it possible to have a set of presets for these options? It could function in the same way that PDF presets work.
8. Separate the three fields in conformance data: EPUB accessibility version (1.0, 1.1), WCAG version (2.0, 2.1, 2.2?), WCAG level (A, AA, AAA)
9. The certified report link has a browse for file link. The report is usually a URL, not an attachable file. 
10. Use English language terms for filed names and values (but use the correct strings for the actual metadata!)
11. Will list out the options for accessModeSufficient, and Hazards in the next comment.

**Comment by LauraB7** [comment]
### AccessModeSufficient

- visual
- textual
- auditory
- tactile
- visual, textual
- visual, auditory
- visual, textual, auditory
- visual, tactile
- visual, textual, tactile
- visual, auditory, tactile
- visual, textual, auditory, tactile
- textual, auditory
- textual, tactile
- textual, auditory, tactile
- auditory, tactile

Note: You can choose multiple of the options above

### Hazards

- none
- noFlashingHazard
- noMotionSimulationHazard
- noSoundHazard
- unknown
- unknownFlashingHazard
- unknownMotionSimulationHazard
- unknownSoundHazard
- flashing
- motionSimulation
- sound

Note: picking an individual option for each of the sound, motionSimulation, and flashing hazard is required or selecting the overarching “none” or “unknown” for all three is common.

**Comment by CircularKen** [comment]
I note that this feature is not present in Adobe InDesign 2024 (19.0)

**Comment by raman211** [comment]
Hi @ways2read, @LauraB7 , @gregoriopellegrino, @CircularKen 

This is Ramanjulu from the Adobe InDesign Team. We are looking to give content creators a way to add the above accessibility and need some clarity from you all.

As per our proposed design specs here https://xd.adobe.com/view/f296cb1f-2874-4e2f-83d4-2cf6fdc06794-a7c5/
We would want to give this feature as part of the December Release.

As you are aware, these values are going to be added to the metadata fields something like 

**<meta property="schema:accessibilityHazard">noFlashingHazard</meta>** and

![image](https://github.com/ways2read/InDesignA11Y/assets/12660080/22c1736f-1497-4804-9d91-0b787ffe5a4e)

We would want to know If the strings that are part of the dropdown lists need translation into different locales or not.

If they need translation, would it make sense to break the values like **noFlashingHazard** for hazard or **pageBreakMarkers** for feature into multiple words?

**Comment by gregoriopellegrino** [comment]
Dear @raman211,
thank you for the exchange.

Our considerations are:

- the UI field labels are translatable, but it's advisable **not to translate** the values within the dropdown list fields as doing so could potentially make it challenging for users to comprehend the metadata content
- for the user interface we suggest splitting the words that are in _camelCase_, e.g. _noFlashingHazard_ becomes _No Flashing Hazard_, _pageBreakMarkers_ becomes _Page Break Markers_, etc. (while in the XML code they should be placed in _camelCase_)
- can you tell us when the pre-release of this version will be available? We would like to test it and give you feedback before official release

Thank you,
Gregorio, Laura and Richard

**Comment by AbhinavAgar** [comment]
Thanks @gregoriopellegrino, Laura and Richard for recommendation.

Quick question:

If UI fields are not translated, they will appear in English strings in non-English builds of InDesign. Will that not be a problem for users of non-English builds? Example, a German user, using German build of InDesign, will see the string "noFlashingHazard"?  Typically, we always translate strings in InDesign unless those are standard strings (like EPUB, JPG etc.). Thus, wanted to double check if the recommendation to not-translate the strings is as per the Accessibility standard?

Thanks
Abhinav Agarwal

**Comment by raman211** [comment]
Hi @gregoriopellegrino,

We would share the pre-release builds by the end of the next week (17th Nov 2023)

Thanks,
Ramanjulu.

**Comment by mattgarrish** [comment]
The terms referenced here come from the vocabulary for the schema.org properties. These values are defined in this document: https://www.w3.org/community/reports/a11y-discov-vocab/CG-FINAL-vocabulary-20230718/

Although English-derived tokens, the values are not necessarily meant to be directly accessed by users in that form, not even English users. They're intended to be translated as needed for human reading, but the tokens themselves don't change based on the user's language (i.e., so every developer can count on finding the same set of tokens in any EPUB and only translate them into their language).

The ideal here, it sounds like, is for users to get language-specific labels, like "No flashing hazard" or "Aucune risque de flashe" to select from, but for the metadata, in this case, to always be `<meta property="accessibilityHazard">noFlashingHazard</meta>`.

Is that possible to do, or does the value the user select always have to be exactly what goes in the meta tag?

**Comment by raman211** [comment]
@mattgarrish @gregoriopellegrino 
Thanks for your inputs, we considered your feedback.
Keeping the InDesign UI paradigms intact, we will be translating all the UI  options (exactly as suggested by @mattgarrish above)
So we will be shipping this feature in 19.2 (likely mid-Jan 2024) instead of 19.1

**Comment by raman211** [comment]
Hi @ways2read, @LauraB7 , @gregoriopellegrino, @CircularKen

We have released this feature as part of the pre release builds a week back and the same are available here with 
https://forums.adobeprerelease.com/indesigndev/discussion/430/rcs-builds-for-indesign-incopy-and-ids-2024-19-0-1-205-along-with-the-sdk-are-now-available#latest
https://forums.adobeprerelease.com/indesign/discussion/2377/rcs-builds-for-indesign-and-incopy-2024-19-0-1-205-builds-are-now-available-to-download#latest

Hope you have played with the feature and we are waiting to hear your feedback on it.

Thanks,
Ramanjulu.

**Comment by ways2read** [comment]
The drop-down list of accessibility features looks great!

**Tagged PDF** and **Bookmarks** are for PDFs and it is confusing if they're in the list for EPUB export. Request these are removed.

With thanks to advice from @mattgarrish we also suggest removing some others:

- **ChemML** - not supported in browsers and is not legal in EPUB as it’s not a recognized extension of HTML
- **Large print** - I think it just adds confusion because we don't think anyone formats EPUBs specifically. 
- **Tactile Object**, **Tactile Graphic**, and **Braille** – if these are included, we suspect it’s not by InDesign users

![image](https://github.com/ways2read/InDesignA11Y/assets/9092859/74dc47cc-4ea1-4c39-b955-75ec9526fe1a)

**Comment by raman211** [comment]
@ways2read we are past the 19.2 release timelines, we will address this issue in 19.3 which is one month away.

**Comment by ways2read** [comment]
Could you please clarify? The accessibility and conformance metadata feature will ship in 19.2 (as it is in the prerelease), and then you can look at the suggested improvements 19.3? Is that correct?

**Comment by raman211** [comment]
@ways2read Yes Richard, We have already shipped 19.2
The above-suggested improvement will be part of 19.3
Thanks,
Ramanjulu

**Comment by ways2read** [comment]
OK sure. Looking forward to receiving 19.2 through the release channel (I guess this will come in the coming days). Then there are some issues we should be able to close. 👍

**Comment by gregoriopellegrino** [comment]
@raman211 the release in version 19.2 satisfies us, for version 19.3 we would ask you to remove the accessibility features as shown in https://github.com/ways2read/InDesignA11Y/issues/19#issuecomment-1908275808

**Comment by CircularKen** [comment]
These extra options are set manually on export and are retained in the file, but in trying to access or influence these settings using scripting I have not been able to find them…

@raman211 Can you please confirm whether they are scriptable?

**Comment by jonaslil** [comment]
Writing on behalf of the group, we are pleased to note that the accessibility features listed in https://github.com/ways2read/InDesignA11Y/issues/19#issuecomment-1908275808 have been removed in the latest prereleases.

Having had a closer look, we request the following properties are also removed:

Annotations - not relevant for epub
Captions - deprecated
Print Page Numbers - no longer recommended, replaced by Page break markers which is included in the drop-down
Timing Control - not relevant for epub

The following one should be added:
Open Captions (= openCaptions)

Synchronized AudioText should be Synchronized Audio Text, in line with the principle of splitting camel case strings into separate words.

**Comment by CircularKen** [comment]
By making this feature unable to be accessed by scripting you are asking the InDesign user to make very complex, unfamiliar and jargon filled choices for each document and only at the time of export.

In a similar way to other complex export topics like colour management and PDF/X exporting the answer for the regular InDesign user is to rely on presets and to expose these to scripting.

Without the ability to set and control these settings on behalf of the user you are only solving this requirement in theory. In reality there will be no use / misuse / incorrect use of this important feature.

Please can you confirm whether you are making this feature able to be applied with presets and that those presets will be scriptable?

**Comment by CircularKen** [comment]
I'm pleased to report that @raman211 has commented over in the InDesign prerelease group that
"These are part of the InDesign 19.3 builds, but not part of the InDesign 19.2" so we can assume they are to be scriptable soon. 👍

**Comment by jonaslil** [comment]
In https://github.com/ways2read/InDesignA11Y/issues/19#issuecomment-1959295083, we suggested removing four accessibility features from the drop-down list and adding one. I'm noting that this has not been fixed in 19.4. Will it be fixed in coming releases?

**Comment by gregoriopellegrino** [comment]
Dear @raman211,
to prepare ourselves for the call scheduled on Monday, June 17th: are there any new updates or releases related to this issue (specifically https://github.com/ways2read/InDesignA11Y/issues/19#issuecomment-1959295083)?

Currently, we are running the _InDesign Pre-Release 19.5 (Mainline CI #f95bd09)_ version on our machines.

Thanks

**Comment by jonaslil** [comment]
As a reminder: the things that still need to be handled in this issue are mentioned in https://github.com/ways2read/InDesignA11Y/issues/19#issuecomment-1959295083

**Comment by jonaslil** [comment]
We're pleased to see that the accessibility features we suggested for removal from the menu have been removed in the latest 20.0 prerelease. However, the addition of Open Captions (openCaptions), which we also recommended, is still missing.

**Comment by jonaslil** [comment]
Noting that Open Captions has been added to the accessibility feature menu in the 20.0.0.55 prerelease build.
